### PR TITLE
[KVM Operator] Performed following code changes

### PR DESCRIPTION
**Pull request by seswarrajan**
1. Included code to extract cluster IP and loadbalancer external IP using k8s core APIs
2. Code cleanup -- Removed unused code
3. Updated docker build files and repository information
4. Updated etcd to use insecure connection, until figuring out the common method for secure TLS connection

### CODE REVIEW

File: operator/build/build_kvmoperator.sh [hunk 1]
```diff
@@ -32,27 +32,27 @@ fi
 # remove old images
 docker images | grep kvmsoperator | awk '{print $3}' | xargs -I {} docker rmi -f {} 2> /dev/null
 
-echo "[INFO] Removed existing kvmsoperator/kvmsoperator images"
+echo "[INFO] Removed existing accuknox/kvmsoperator images"
```
Comment (nyrahul) [inline operator/build/build_kvmoperator.sh:35]:
```suggestion
echo "[INFO] Removed existing kubearmor/kvmsoperator images"
```

File: operator/build/build_kvmoperator.sh [hunk 2]
```diff
@@ -32,27 +32,27 @@ fi
...
+echo "[INFO] Removed existing source files"
 
 # copy files to build
 $ARMOR_HOME/build/copy_source_files.sh
 
 echo "[INFO] Copied new source files"
 
 # build a new image
-echo "[INFO] Building kvmsoperator/kvmsoperator:$VERSION"
+echo "[INFO] Building accuknox/kvmsoperator:$VERSION"
```
Comment (nyrahul) [inline operator/build/build_kvmoperator.sh:48]:
```suggestion
echo "[INFO] Building kubearmor/kvmsoperator:$VERSION"
```

File: operator/build/build_kvmoperator.sh [hunk 3]
```diff
@@ -32,27 +32,27 @@ fi
...
 docker build -t kvmsoperator/kvmsoperator:$VERSION  . -f $ARMOR_HOME/build/Dockerfile.kvmsoperator
```
Comment (nyrahul) [inline operator/build/build_kvmoperator.sh:49]: No point in keeping kvmsoperator in a different repo. Lets push this image in the kubearmor docker id.

File: operator/build/build_kvmoperator.sh [hunk 4]
```diff
@@ -32,27 +32,27 @@ fi
...
 
 if [ $? != 0 ]; then
-    echo "[FAILED] Failed to build kvmsoperator/kvmsoperator:$VERSION"
+    echo "[FAILED] Failed to build accuknox/kvmsoperator:$VERSION"
```
Comment (nyrahul) [inline operator/build/build_kvmoperator.sh:52]:
```suggestion
    echo "[FAILED] Failed to build kubearmor/kvmsoperator:$VERSION"
```

File: operator/build/build_kvmoperator.sh [hunk 5]
```diff
@@ -32,27 +32,27 @@ fi
...
     exit 1
 else
-    echo "[PASSED] Built kvmsoperator/kvmsoperator:$VERSION"
+    echo "[PASSED] Built accuknox/kvmsoperator:$VERSION"
```
Comment (nyrahul) [inline operator/build/build_kvmoperator.sh:55]:
```suggestion
    echo "[PASSED] Built kubearmor/kvmsoperator:$VERSION"
```

File: service/build/build_kvmservice.sh [hunk 6]
```diff
@@ -32,7 +32,7 @@ fi
 # remove old images
 docker images | grep kvmservice | awk '{print $3}' | xargs -I {} docker rmi -f {} 2> /dev/null
 
-echo "[INFO] Removed existing kvmservice/kvmservice images"
+echo "[INFO] Removed existing accuknox/kvmservice images"
```
Comment (nyrahul) [inline service/build/build_kvmservice.sh:35]:
```suggestion
echo "[INFO] Removed existing kubearmor/kvmservice images"
```

File: service/build/build_kvmservice.sh [hunk 7]
```diff
@@ -45,14 +45,14 @@ $ARMOR_HOME/build/copy_source_files.sh
 echo "[INFO] Copied new source files"
 
 # build a new image
-echo "[INFO] Building kvmservice/kvmservice:$VERSION"
-docker build -t kvmservice/kvmservice:$VERSION  . -f $ARMOR_HOME/build/Dockerfile.kvmservice
+echo "[INFO] Building accuknox/kvmservice:$VERSION"
```
Comment (nyrahul) [inline service/build/build_kvmservice.sh:48]:
```suggestion
echo "[INFO] Building kubearmor/kvmservice:$VERSION"
```

File: service/build/build_kvmservice.sh [hunk 8]
```diff
@@ -45,14 +45,14 @@ $ARMOR_HOME/build/copy_source_files.sh
...
+docker build -t accuknox/kvmservice:$VERSION  . -f $ARMOR_HOME/build/Dockerfile.kvmservice
 
 if [ $? != 0 ]; then
-    echo "[FAILED] Failed to build kvmservice/kvmservice:$VERSION"
+    echo "[FAILED] Failed to build accuknox/kvmservice:$VERSION"
```
Comment (nyrahul) [inline service/build/build_kvmservice.sh:52]:
```suggestion
    echo "[FAILED] Failed to build kubearmor/kvmservice:$VERSION"
```

File: service/build/build_kvmservice.sh [hunk 9]
```diff
@@ -45,14 +45,14 @@ $ARMOR_HOME/build/copy_source_files.sh
...
     exit 1
 else
-    echo "[PASSED] Built kvmservice/kvmservice:$VERSION"
+    echo "[PASSED] Built accuknox/kvmservice:$VERSION"
```
Comment (nyrahul) [inline service/build/build_kvmservice.sh:55]:
```suggestion
    echo "[PASSED] Built kubearmor/kvmservice:$VERSION"
```

File: service/core/kvmService.go [hunk 10]
```diff
@@ -73,6 +77,43 @@ type KVMS struct {
 	WgDaemon sync.WaitGroup
 }
 
+func getExternalIP() (string, error) {
+
+	/* Calculated time manually to see that the kvmsoperator service
+	 * takes a minimum of 45 seconds to fetch the same.
+	 * Hence placing a time delay of 1 minute
+	 */
+	time.Sleep(1 * 60 * time.Second)
```
Comment (nyrahul) [inline service/core/kvmService.go:86]: Why a sleep of 1 minute!

File: service/etcd/etcdClient.go [hunk 11]
```diff
@@ -25,30 +24,67 @@ type EtcdClient struct {
 	leaseResponse *clientv3.LeaseGrantResponse
 }
 
+func getEtcdEndPoint() string {
```
Comment (nyrahul) [inline service/etcd/etcdClient.go:27]: This code is complete duplicate of code in `operator/etcd/etcdClient.go`!

**Review by nyrahul — CHANGES_REQUESTED** [review]
* there seems to be code duplication between service and operator code.
* in many places we are using accuknox reference in the log.. pls remove it.

File: operator/build/build_kvmoperator.sh (hunk 1 above)
Comment (seswarrajan) [reply operator/build/build_kvmoperator.sh:35]: But will we be using kubearmor repo or accuknox repo ?

File: operator/build/build_kvmoperator.sh (hunk 2 above)
Comment (seswarrajan) [reply operator/build/build_kvmoperator.sh:48]: Which dockerhub will be used? Kubearmor or accuknox dockerhub ?

File: operator/build/build_kvmoperator.sh (hunk 3 above)
Comment (seswarrajan) [reply operator/build/build_kvmoperator.sh:49]: Sure Rahul

File: operator/build/build_kvmoperator.sh (hunk 2 above)
Comment (nyrahul) [reply operator/build/build_kvmoperator.sh:48]: we have kubearmor dockerhub that we can use.

File: service/etcd/etcdClient.go (hunk 11 above)
Comment (seswarrajan) [reply service/etcd/etcdClient.go:27]: Yes Rahul. But both operator and service are all together different sources and builds. Hence duplicated twice. If duplication needs to be avoided, then we need to place them under KVMService and use them.

**Review by nyrahul — APPROVED** [review]
This is a big PR. Best would be to have some system tests built and verified. I am approving this in the hope that we have tested this manually.

**Comment by nyrahul** [comment]
Looks like all go-* utils are failing. Please handle those changes in workflows.

File: deployments/etcd.yml [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+---
+
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    app: etcd
+    etcd_node: etcd0
+  name: etcd0
+  namespace: kube-system
```
Comment (nyrahul) [inline deployments/etcd.yml:24]: it would be better to not deploy etcd in kube-system nameespace.

**Review by nyrahul — APPROVED** [review]

**Pull request merged**